File: gfg/indexes_Of_sub_array_sum/indexOfSubArraySum.py
def subarraySum(arr,target):
        size = len(arr)
        for i in range(size):
            sum = 0
            for start in range(i,size):
                sum += arr[start]
                if sum == target :
                    return [i+1, start+1]
                elif sum > target :
                    break
        return -1

if __name__ == "__main__":
        arr = [1,2,5,6,3,8]
        target = 3
        result = subarraySum(arr, target)
        print("Actual Array : ", arr)
        print("Target :", target)
        if result == -1:
                print(-1)
        else:
                print(result, sep=", ")

File: gfg/indexesOfSubArray/indexOfSubArraySum.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> subarraySum(vector<int> &arr, int target) {

        int size = arr.size();
        int sum;
        for(int i = 0; i < size; i++ ){
            sum = 0;
            for (int start = i; start < size; start++){
                sum += arr.at(start);
                if (sum == target){
                    return {i+1, start+1};
                }else if(sum > target){
                    break;
                }
            }
        }
        return {-1};
}

int main(){
	// Sample Code for output
	vector <int> arr = {11,7,4,2,8,6,3,1};
	int target = 9;
	vector <int> result;
	cout << "Actual Array :";
	for(int value : arr){
		cout << value << " ";
	}
	cout << "\nTarget :" << target << endl;
	result = subarraySum(arr, target);
	if (result[0] == -1){
		cout << -1 << endl;
	}else{
		cout << "Result : [" << result[0] << ", " << result[1] << "]" << endl ;
	}
	
	return 0;
}
